pkg/loop/internal/pb/evm/evm.proto

```diff
@@ -123,6 +123,74 @@ message Head {
   Hash parent_hash = 4;
 }
 
+message Expression {
+ oneof evaluator {
+    EXTPrimitive primitive = 1;
+    BooleanExpression boolean_expression = 2;
+  }
+}
+
+enum BooleanOperator {
+  AND = 0;
+  OR = 1;
+}
+
+message BooleanExpression {
+  BooleanOperator boolean_operator = 1;
+  repeated Expression expression = 2;
+}
+
+message And {
+  repeated Expression expr = 1;
+}
+
+message Or {
+  repeated Expression expr = 1;
+}
+
+
+// EXTPrimitive extends general primitives with evm-specific primitives 
+message EXTPrimitive {
+  oneof primitive{
+	Primitive general_primitive = 1;
+	EVMPrimitive evm_primitive = 2;
+  }
+}
+
+message HashValueComparator {
+  repeated Hash values = 1;
+  ComparisonOperator operator = 2;
+}
+
+message AddressFilter {
+  Address address = 1;
+}
+
+message EventSigFilter {
+  Hash eventSig = 1;
+}
+
+message EventByWordFilter {
+  uint32 word_index = 1;
+  repeated HashValueComparator  hashed_value_comparers = 2;
+}
+
+message EventByTopicFilter {
+  uint64 topic = 1;
+  repeated HashValueComparator hashed_value_comparers = 2;
+}
+
+// evm-style primitives
+// can be used in combination with general primnitives for querying tracked logs
+message EVMPrimitive {
```

Comment:
Let's drop the prefix since it's in `package evm`:
```suggestion
message Primitive {
```

Reply:
Actually, it seems like we need some other qualifier since these are not general primitives at all. What are we trying to capture here? A union of query filters?

Reply:
> I can't use Primitive name since it will override general Primitive from contract_reader

Names outside of this package are not a collision because they are package qualified